src/generic/treectlg.cpp

```diff
@@ -1012,18 +1005,16 @@ void wxGenericTreeCtrl::EnableBellOnNoMatch( bool on )
     m_findBell = on;
 }
 
-void wxGenericTreeCtrl::OnSysColourChanged(wxSysColourChangedEvent&)
+void wxGenericTreeCtrl::InitVisualAttributes()
 {
     const wxVisualAttributes attr(GetDefaultAttributes());
     SetOwnForegroundColour(attr.colFg);
     SetOwnBackgroundColour(attr.colBg);
-    if (!m_hasExplicitFont)
+    if (!m_hasFont)
```

Comment:
`m_hasFont` will always be true after `SetOwnFont()` is called with a valid font, so subsequent executions of this code may not have the same effect as the first. That's why I had to add `m_hasExplicitFont`.

BTW, why is it necessary to set the font and foreground/background colors here at all? Won't these values be the defaults anyway?

Reply:
Oops, yes, of course, thanks for the explanation. A trivial fix would be to add `m_hasFont = false;` after `SetOwnFont()` call because we know that it's not _really_ set. I also don't really understand why do we do this check for the font but not the colours and [the original patch](https://trac.wxwidgets.org/ticket/6738) doesn't really explain it neither...

We do need to set the colours/fonts because we want them to default to those from `GetClassDefaultAttributes()` and not the defaults inherited from the parent.